lua/entities/sammyservers_textscreen/cl_init.lua

```diff
@@ -73,10 +67,48 @@ end
 function ENT:Think()
 end
 
+function ENT:OnRemove()
+	screenInfo[self] = nil
+end
+
 net.Receive("textscreens_update", function(len)
 	local ent = net.ReadEntity()
 
 	if IsValid(ent) and ent:GetClass() == "sammyservers_textscreen" then
-		ent.lines = net.ReadTable()
+
+		local t = net.ReadTable()
+		local t2 = {}
+		local curheight = 0
+		local totheight = 0
+
+		ent.lines = t
+
+		for i=1, #t do
+			t2[i] = {}
+			// Text
+			t2[i].text = t[i].text
+			// Colour
+			t2[i].color = t[i].color
+			// Font
+			t2[i].font = (validFont(t[i].font) or textscreenFonts[1]) .. t[i].size
+			// Textsize
+			surface.SetFont(t2[i].font)
+			local TextWidth, TextHeight = surface.GetTextSize(t2[i].text)
+			// Pos
+			totheight = totheight + TextHeight
+			t2[i].height = TextHeight
+		end
+
+		for i=1, #t do
+			t2[i].pos = -(totheight / 2) + curheight
+			curheight = curheight + t2[i].height
+		end
+
+		t2.tableSize = #t
+
+//		PrintTable(t2)
```

Comment:
By the time you're submitting a pull request, you should try to remove all commented code.

If left in, they tend to crop up over the years, while nobody remembers why it's there, who put it there, or if it's safe to remove.